src/parser/expression.js

```diff
@@ -1072,8 +1105,27 @@ export default class ExpressionParser extends LValParser {
     return false;
   }
 
+  parseFunctionReturnTypeAndBodyAndFinish(node: N.BodilessFunctionOrMethodBase, type: string, allowExpressionBody?: boolean): void {
```

Comment:
I think this should just be `typescript.js#parseFunctionBodyAndFinish` which overwrites `expression.js#parseFunctionBodyAndFinish`